### DISCUSSION TITLE
Github Action Eslint problem

### DISCUSSION THREAD

**Discussion by AlexMachin1997** [community/community#25378] [Actions]
<p>Hi,</p>
<p>Im setting up a Github action to identify any linting issues and automatically fix them, but Im having some trouble with the action.</p>
<p>It keeps running this “Unknown event” after each Github action, see below image.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/151df0d0d5895143129c455001a966ec323fbe6f.png" data-download-href="/uploads/short-url/30OaONjzEyfOlJPtnGmWsYFxCZ9.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29917195/181090695-56825095-5a20-4dae-b023-7ef76acb7a56.png" alt="image" data-base62-sha1="30OaONjzEyfOlJPtnGmWsYFxCZ9" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/1/151df0d0d5895143129c455001a966ec323fbe6f_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/2X/1/151df0d0d5895143129c455001a966ec323fbe6f_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/1/151df0d0d5895143129c455001a966ec323fbe6f_2_1380x776.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/151df0d0d5895143129c455001a966ec323fbe6f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1080 295 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’ve also noticed the action is poiting out I have a number of issues with my code, see image below</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/64b91c20c6f7514bbd205c252c178973ec6006f7.png" data-download-href="/uploads/short-url/en2lgXTlWJ5lU1XGSqrkixQ8aPl.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29917195/181090696-c9074b2c-53ed-4866-a9d6-1220e8bc3eee.png" alt="image" data-base62-sha1="en2lgXTlWJ5lU1XGSqrkixQ8aPl" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/64b91c20c6f7514bbd205c252c178973ec6006f7_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/2X/6/64b91c20c6f7514bbd205c252c178973ec6006f7_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/6/64b91c20c6f7514bbd205c252c178973ec6006f7_2_1380x776.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/64b91c20c6f7514bbd205c252c178973ec6006f7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1080 302 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>But this seems odd, it’s almost like it’s taking into account node_modules.</p>
<p>Could someone lend a hand please :slight_smile:</p>
<p>Repository link: <a href="https://github.com/AlexMachin1997/TheOpenMovieDB-React-Example" rel="nofollow noopener">https://github.com/AlexMachin1997/TheOpenMovieDB-React-Example</a></p>
<p>Weird errors which seem to break the action completely :confused:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/be88200a9dd428c2e1f7c579c3f5213a83cbfee1.png" data-download-href="/uploads/short-url/rbwmHFFN6XOgicETPuRC0gl522R.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29917195/181090697-1c5df477-5479-437f-ae86-535b3b941da0.png" alt="image" data-base62-sha1="rbwmHFFN6XOgicETPuRC0gl522R" width="690" height="249" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/be88200a9dd428c2e1f7c579c3f5213a83cbfee1_2_690x249.png, https://aws1.discourse-cdn.com/github/optimized/2X/b/be88200a9dd428c2e1f7c579c3f5213a83cbfee1_2_1035x373.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/b/be88200a9dd428c2e1f7c579c3f5213a83cbfee1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/be88200a9dd428c2e1f7c579c3f5213a83cbfee1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1356×490 22.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/bb575231d134f68037c8efad33e5fd11e246f942.png" data-download-href="/uploads/short-url/qJin254gEUjN4bDB1U03oVWw2Km.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29917195/181090698-787e9d1e-52a8-45c4-a42b-8c4674bdb51e.png" alt="image" data-base62-sha1="qJin254gEUjN4bDB1U03oVWw2Km" width="690" height="388" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/bb575231d134f68037c8efad33e5fd11e246f942_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1128×635 33 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jeffrafter** [comment, marked as answer]
<p>It seems like the problems are related: the action is fixing things in node_modules and then trying to commit them but node_modules are ignored… so there is nothing to commit. I’m not sure if this is right, but node_modules should be ignored by default in eslint. You eslintignore spec feels off though:</p>
<p><code>./node_modules/**/**/*.js</code></p>
<p>Should just be:</p>
<p><code>node_modules/*</code></p>
<p>Just ignore everything in node modules. Additionally you might want to add a <code>.prettierignore</code> with the same.</p>

**Comment by AlexMachin1997** [comment]
<p>Hi Heffrafter,</p>
<p>I added the prettier ignore and and adjusted the eslintignore but Im still getting this error.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/867c90e4288957278357a29477ac045431837c06.png" data-download-href="/uploads/short-url/jbIRt9UvAgdSBsvkgMXaRuEnj1Q.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29917195/181090699-57822496-c316-4d62-9acb-a3275123d0ed.png" alt="image" data-base62-sha1="jbIRt9UvAgdSBsvkgMXaRuEnj1Q" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/867c90e4288957278357a29477ac045431837c06_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/867c90e4288957278357a29477ac045431837c06_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/8/867c90e4288957278357a29477ac045431837c06_2_1380x776.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/867c90e4288957278357a29477ac045431837c06_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1080 291 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by AlexMachin1997** [comment]
<p>So I got it to work by removing the custom commit message, I beleive that was causing the issue.</p>
<p>I do have one concern a weird unkown event runs after the actions have completed.</p>
<p>The error contradicts itself, I want single quotes but it wants double WTF :laughing: so confused why it’s doing this.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/95f986349a4332b6da4a7ea732f650c21a49909e.png" data-download-href="/uploads/short-url/loJKgPY40M2i3Nb40qGzuLJerjo.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29917195/181090701-19b6b1ac-6684-4306-89fe-f0875c64ea46.png" alt="image" data-base62-sha1="loJKgPY40M2i3Nb40qGzuLJerjo" width="690" height="110" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/95f986349a4332b6da4a7ea732f650c21a49909e_2_690x110.png, https://aws1.discourse-cdn.com/github/optimized/2X/9/95f986349a4332b6da4a7ea732f650c21a49909e_2_1035x165.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/9/95f986349a4332b6da4a7ea732f650c21a49909e_2_1380x220.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/95f986349a4332b6da4a7ea732f650c21a49909e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1454×233 9.84 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I fixed the quotes issue after finding this <a href="https://github.com/prettier/prettier/issues/2280" rel="nofollow noopener">https://github.com/prettier/prettier/issues/2280</a>. Im now getting decent error messages which I can fix in the morning. Linting + auto fixing works :slight_smile:</p>
